### DISCUSSION TITLE
Deploy to GitHub-Pages via Pipeline or Action?

### DISCUSSION THREAD

**Discussion by suchja** [statiqdev/Discussions#221] [Q&A] [answered]
I would like to deploy a small demo project for a theme to GitHub Pages (see [suchja/statiq-dopetrope-demo](https://github.com/suchja/statiq-dopetrope-demo). Now I came across [statiq's official deployment guide GitHub Pages](https://www.statiq.dev/guide/deployment/github-pages) and understood that statiq should be run via GitHub Actions, but that it includes its own deployment pipeline for GitHub Pages. I tried that way many times with different settings, permissions, ... but can't get it working.

Now looking into other repositories utilizing statiq to publish a GitHub Page I found the way to purely use GitHub Actions to deploy the contents. So basically it means that statiq is used to generate the static site, but its deployment pipeline is NOT used. Instead the appropriate actions are used.

Which is the "better" way and why is the pipeline way not working???

**Comment by daveaglick** [comment, marked as answer]
There's definitely not a "better" way and both are totally valid. I personally like to keep _everything_ in CI/CD scenarios contained in a single command line application and have that do the driving, so the use of a deploy phase where Statiq itself uploads to GitHub Pages comes from that personal preference. That said, there are a lot of reasons why leveraging built-in actions/steps/tasks that are native to the CI/CD environment could be preferred - for example, their use might enable some interesting functionality beyond what's available through Statiq, or allow deployment to locations that Statiq doesn't yet support (like Azure Static Web Apps or Cloudflare Pages).

As for the problems with GitHub Pages deployment via Statiq, it looks like maybe the discussion happening in https://github.com/orgs/statiqdev/discussions/148 is related? If not, let me know and I'll dive a little deeper.

**Reply by suchja** [reply]
@daveaglick thank you so much for the feedback and Time!

I understand the point regarding the "better" way (and agree with you). You are right https://github.com/orgs/statiqdev/discussions/148 is related to a degree. However, when using the pipeline deployment path (your preferred way), the permission problem as discussed in https://github.com/statiqdev/Statiq.Web/issues/1021 seems not to be the only issue. I prepared a [simple test-repro](https://github.com/suchja/statiq-cleanblog-demo) based on your cleanblog-Theme (in order to avoid problems from my theme).

Now the first misunderstanding was, that the pipeline based deployment uses a GitHub Action for triggering the build, BUT GitHub Pages are sourced by the branch and NOT the GitHub Action. It simply wasn't clear to me that the deployment pipeline pushes everything to a branch. At least for newbies this should be documented. While "debugging" I even tried the main branch and suddenly had the final page in the main branch instead of the source.
If I understand it correctly, the pipeline pushes the created site from output folder to a branch, BUT the "new?" approach GitHub follows with its Actions is that a GitHub Page is deployed somewhere as an artifact and thus the deployed site is not available in any branch at all.
This is all to say a bit more documentation would be highly appreciated. I'll prepare a PullRequest for the Guide.

Finally it seems something is not properly working with `host`, `LinkRoot` and paths. If you have a look on the [deployed GitHub Pages](https://suchja.github.io/statiq-cleanblog-demo/) of my [cleanblog demo](https://github.com/suchja/statiq-cleanblog-demo), you'll see that all resources are not found.
I found this discussion https://github.com/orgs/statiqdev/discussions/56 , but not sure whether that is really the problem.

**Reply by suchja** [reply]
> Finally it seems something is not properly working with host, LinkRoot and paths. If you have a look on the [deployed GitHub Pages](https://suchja.github.io/statiq-cleanblog-demo/) of my [cleanblog demo](https://github.com/suchja/statiq-cleanblog-demo), you'll see that all resources are not found.
> I found this discussion https://github.com/orgs/statiqdev/discussions/56 , but not sure whether that is really the problem.

My bad! I wasn't aware that `host` is really only the host and that `LinkRoot` in the "sub-path". So initially I had `Host: suchja.github.io/statiq-cleanblog-demo`. After removing `statiq-cleanblog-demo` from `Host` and only using it via `LinkRoot`, everything works perfectly.

**Reply by AlexHedley** [reply]
https://suchja.github.io/statiq-cleanblog-demo/ looks great.

Are you missing some styles on https://suchja.github.io/statiq-dopetrope-demo/?

**Reply by suchja** [reply]
> https://suchja.github.io/statiq-cleanblog-demo/ looks great.

Thanks 🙏!

> Are you missing some styles on https://suchja.github.io/statiq-dopetrope-demo/?

Maybe. I was playing around with the appsettings.json as described in https://github.com/orgs/statiqdev/discussions/222 Thus the latest version might be a one with some false settings.
